spec/core/ClockSpec.js

```diff
@@ -689,9 +696,86 @@ describe('Clock (acceptance)', function() {
     expect(recurring1.calls.count()).toBe(4);
   });
 
+  it('can run setTimeouts/setIntervals asynchronously', function() {
```

Comment:
@coderabbitai rewrite this function to use ES6 or newer Javascript features. Focus on performance.